torch/functional.py

```diff
@@ -1465,8 +1464,7 @@ def _lu_impl(A, pivot=True, get_infos=False, out=None):
        ``L``, ``U``, and ``P`` can be derived using :func:`torch.lu_unpack`.
 
     .. warning::
-        The LU factorization does have backward support,
-        but only for square inputs of full rank.
+        The backward step for the LU factorization is stable only for inputs of full rank!
```

Comment:
```suggestion
        The gradient (or derivative) computation for the LU factorization is stable only for inputs of full rank!
```

Reply:
CC @albanD  to clarify better terminology.

Reply:
In `torch.linalg` we refer to these as "Gradient computations".
Following the notation in `torch.linalg` (see [QR](https://pytorch.org/docs/1.9.0/generated/torch.linalg.qr.html) and [SVD](https://pytorch.org/docs/stable/generated/torch.linalg.svd.html)) this could be written as:
> Gradients computed using `L` or `U` will only be finite when `A` is full rank. This is because the LU decomposition is just differentiable at matrices of full rank. Furthermore, if `A` is close to not being full rank, the gradient will be numerically unstable as it depends on the computation of `L^{-1}` and `U^{-1}`.